encodings/fastlanes/src/delta/serde.rs

```diff
@@ -3,20 +3,21 @@ use vortex_array::validity::Validity;
 use vortex_array::vtable::EncodingVTable;
 use vortex_array::{
     Array, ArrayChildVisitor, ArrayContext, ArrayRef, ArrayVisitorImpl, DeserializeMetadata,
-    EncodingId, RkyvMetadata,
+    EncodingId, ProstMetadata,
 };
 use vortex_dtype::{DType, PType, match_each_unsigned_integer_ptype};
 use vortex_error::{VortexExpect, VortexResult, vortex_bail};
 
 use super::DeltaEncoding;
 use crate::DeltaArray;
 
-#[derive(Debug, Clone, rkyv::Archive, rkyv::Serialize, rkyv::Deserialize)]
+#[derive(Clone, prost::Message)]
 #[repr(C)]
 pub struct DeltaMetadata {
-    // TODO(ngates): do we need any of this?
+    #[prost(uint64, tag = "1")]
     deltas_len: u64,
-    offset: u16, // must be <1024
+    #[prost(uint32, tag = "2")]
+    offset: u32, // must be <1024
```

Comment:
is it fiddly to keep this as u16 in code (I know protobuf uses variable length encoding anyway)

Reply:
This is just the serialized form. I think it's nice to have the tightest possible bound